arangod/Cluster/ClusterRepairOperations.h

```diff
@@ -0,0 +1,376 @@
+
+// Parameters used in Operation-constructors
+
+namespace tag {
+struct database;
+struct collectionId;
+struct collectionName;
+struct protoCollectionId;
+struct protoCollectionName;
+struct shard;
+struct protoShard;
+struct from;
+struct to;
+struct isLeader;
+struct protoReplicationFactor;
+struct collectionReplicationFactor;
+struct replicationFactor;
+struct renameDistributeShardsLike;
+struct shards;
+struct leader;
+struct followers;
+struct protoFollowers;
+struct distributeShardsLike;
+struct repairingDistributeShardsLike;
+struct shardsById;
+struct deleted;
+struct isSmart;
+}
+namespace {
+keyword<tag::database, std::string> _database = decltype(_database)::instance;
+keyword<tag::collectionId, std::string> _collectionId =
+    decltype(_collectionId)::instance;
+keyword<tag::collectionName, std::string> _collectionName =
+    decltype(_collectionName)::instance;
+keyword<tag::protoCollectionId, std::string> _protoCollectionId =
+    decltype(_protoCollectionId)::instance;
+keyword<tag::protoCollectionName, std::string> _protoCollectionName =
+    decltype(_protoCollectionName)::instance;
+keyword<tag::shard, std::string> _shard = decltype(_shard)::instance;
+keyword<tag::protoShard, std::string> _protoShard =
+    decltype(_protoShard)::instance;
+keyword<tag::from, std::string> _from = decltype(_from)::instance;
+keyword<tag::to, std::string> _to = decltype(_to)::instance;
+keyword<tag::isLeader, bool> _isLeader = decltype(_isLeader)::instance;
+keyword<tag::protoReplicationFactor, size_t> _protoReplicationFactor =
+    decltype(_protoReplicationFactor)::instance;
+keyword<tag::collectionReplicationFactor, size_t> _collectionReplicationFactor =
+    decltype(_collectionReplicationFactor)::instance;
+keyword<tag::replicationFactor, size_t> _replicationFactor =
+    decltype(_replicationFactor)::instance;
+keyword<tag::renameDistributeShardsLike, bool> _renameDistributeShardsLike =
+    decltype(_renameDistributeShardsLike)::instance;
+keyword<tag::shards, std::vector<ShardWithProtoAndDbServers>> _shards =
+    decltype(_shards)::instance;
+keyword<tag::leader, std::string> _leader = decltype(_leader)::instance;
+keyword<tag::followers, std::vector<std::string>> _followers =
+    decltype(_followers)::instance;
+keyword<tag::protoFollowers, std::vector<std::string>> _protoFollowers =
+    decltype(_protoFollowers)::instance;
+keyword<tag::distributeShardsLike, boost::optional<CollectionID>>
+    _distributeShardsLike = decltype(_distributeShardsLike)::instance;
+keyword<tag::repairingDistributeShardsLike, boost::optional<CollectionID>>
+    _repairingDistributeShardsLike =
+        decltype(_repairingDistributeShardsLike)::instance;
+keyword<tag::shardsById, std::map<ShardID, DBServers, VersionSort>>
+    _shardsById = decltype(_shardsById)::instance;
+keyword<tag::deleted, bool> _deleted = decltype(_deleted)::instance;
+keyword<tag::isSmart, bool> _isSmart = decltype(_isSmart)::instance;
+}
+
+struct BeginRepairsOperation {
```

Comment:
Can you add plain text descriptions what those operations are supposed to do and which preconditions they yield either/or/and in `.cpp` or `.h` above the Functions.